### PR TITLE
Jinho/fixed main 1

### PR DESCRIPTION
**Pull request by zhkim216**
"Fixed issue with stage 2 sampling not preserving the backbone, and added support for motif scaffolding with new configurations"

### CODE REVIEW

File: draw_samples.py [hunk 1]
```diff
@@ -41,15 +41,20 @@ def draw_and_save_samples(
     **sampling_kwargs,
 ):
     device = model.device
+    motif_scaffolding = sampling_kwargs['motif_scaffolding'] #* 241101 bug fixed
```
Comment (alexechu) [inline draw_samples.py:44]: comment can be removed

File: draw_samples.py [hunk 2]
```diff
@@ -117,7 +124,7 @@ def __init__(self):
         self.parser.add_argument(
             "--sampling_configdir",
             type=str,
-            default="configs/sampling.yml",
+            default="configs/uncond_sampling.yml", #* 241101 bug fixed
```
Comment (alexechu) [inline draw_samples.py:127]: comment can be removed

File: draw_samples.py [hunk 3]
```diff
@@ -143,31 +150,57 @@ def __init__(self):
             type=int,
             required=False,
             help="If steplen not provided, how many random lengths to sample at",
-        )
+        )
```
Comment (alexechu) [inline draw_samples.py:153]: have you run `black` formatting yet?

File: draw_samples.py [hunk 4]
```diff
@@ -143,31 +150,57 @@ def __init__(self):
...
+            "--motif_scaffolding",
+            type=bool,
+            default=False,
+            help="whether to use motif scaffolding generation", #* 241101 bug fixed
+        )   
         self.parser.add_argument(
             "--input_pdb",
             type=str,
-            default="samples/len100_samp1.pdb",
+            default=None, #* 241101 bug fixed
```
Comment (alexechu) [inline draw_samples.py:166]: comment cna be removed

File: draw_samples.py [hunk 5]
```diff
@@ -143,31 +150,57 @@ def __init__(self):
...
+            default=None,
+            required=False,
+            help="number of samples for conditional generation", #* 241101 bug fixed
+        )                
         self.parser.add_argument(
             "--resample_idxs",
             type=str,
             required=False,
-            default="20-99",
+            default=None, #* 241101 bug fixed
```
Comment (alexechu) [inline draw_samples.py:181]: same

File: draw_samples.py [hunk 6]
```diff
@@ -143,31 +150,57 @@ def __init__(self):
...
             required=False,
             help="PDB file to condition on",
         )
+        self.parser.add_argument(
+            "--cond_num_samples",
+            type=int,
+            default=None,
+            required=False,
+            help="number of samples for conditional generation", #* 241101 bug fixed
```
Comment (alexechu) [inline draw_samples.py:175]: same

File: draw_samples.py [hunk 7]
```diff
@@ -180,10 +213,16 @@ def main():
     print(args)
     is_test_run = False
     seed = 0
-    samples_per_len = args.perlen
-    min_len = args.minlen
-    max_len = args.maxlen
-    len_step_size = args.steplen
+    
+    #* 241101 bug fixed
```
Comment (alexechu) [inline draw_samples.py:217]: same

File: models.py [hunk 8]
```diff
@@ -721,9 +723,10 @@ def design_with_fullmpnn(
             ]
             xt = gt_coords_traj[0]
 
-        gt_atom_mask = None
-        if gt_aatype is not None:
-            gt_atom_mask = utils.atom37_mask_from_aatype(gt_aatype, seq_mask)
+        #* commented out to fix bugs
```
Comment (alexechu) [inline models.py:726]: same

File: my_pdb.pdb
Comment (alexechu) [inline my_pdb.pdb:1]: what is this file for?

**Review by alexechu — COMMENTED** [review]
Thanks Jinho! Mostly looks good. Please add a more descriptive PR title and description.

File: my_pdb.pdb
Comment (zhkim216) [reply my_pdb.pdb:1]: It's an example pdb

**Review by alexechu — APPROVED** [review]
Thanks Jinho, looks great! Please provide a more descriptive PR title, and rename my_pdb.pdb to something like example_pdb.pdb or example.pdb before merging.

**Pull request merged**